app/_lib/connect/token.ts

```diff
@@ -11,22 +13,32 @@ interface GetAccessTokenParams {
 }
 
 export default async function getAccessToken(
-  { code, redirectUri, clientId, clientSecret }: GetAccessTokenParams):
-  Promise<any> {
+  { code, redirectUri, clientId, clientSecret }: GetAccessTokenParams
+): Promise<any> {
   const bodyParams = new URLSearchParams({
     grant_type: GRANT_TYPE,
     code,
-    redirect_uri: redirectUri
+    redirect_uri: redirectUri,
+    client_id: clientId
   })
 
-  try {
+  if (clientId !== ARMT_CLIENT_ID) {
+    bodyParams.append("client_secret", clientSecret)
+  }
+
+  const headers: Record<string, string> = {
+    "Content-Type": "application/x-www-form-urlencoded",
+  }
+
+  if (clientId !== "aRMT") {
```

Comment:
@callumstew Also removed auth for aRMT because Hydra doesn't support empty secret and the default client secret for aRMT is empty.